tests/server/orchestration/api/test_concurrency_limits_v1_adapter.py

```diff
@@ -0,0 +1,493 @@
+        "/concurrency_limits/", json={"tag": "bar", "concurrency_limit": 3}
+    )
+    assert response.status_code == 201
+    initial_id = response.json()["id"]
+
+    # Create again with different limit (should update)
+    response = await client.post(
+        "/concurrency_limits/", json={"tag": "bar", "concurrency_limit": 7}
+    )
+    assert response.status_code == 200  # Not 201 since it's an update
+    data = response.json()
+    assert data["id"] == initial_id  # Same ID
+    assert data["concurrency_limit"] == 7  # Updated limit
+
+
+async def test_v1_create_returns_v1_id(client: AsyncClient, session):
+    """The adapter should return the mirrored V1 identifier, not the V2 id."""
+    response = await client.post(
+        "/concurrency_limits/", json={"tag": "idcheck", "concurrency_limit": 2}
+    )
+    assert response.status_code == 201
+    payload = response.json()
+
+    async with session.begin():
+        v2 = await v2_models.read_concurrency_limit(session=session, name="tag:idcheck")
+        v1 = await v1_models.read_concurrency_limit_by_tag(
+            session=session, tag="idcheck"
+        )
+
+    assert v1 is not None
+    assert payload["id"] == str(v1.id)
+    assert v2 is not None
+    assert str(v2.id) != payload["id"]
+
+
+async def test_v1_read_by_tag_uses_v2_leases(client: AsyncClient, session, monkeypatch):
+    """Test that V1 READ endpoint populates active_slots from V2 leases."""
+    # Create V2 limit directly in DB for tag:foo
+    v2 = await create_concurrency_limit(
+        session=session,
+        concurrency_limit=ConcurrencyLimitV2(name="tag:foo", limit=5),
+    )
+    await session.commit()
+
+    # Stub lease storage to return a lease with a task_run holder for this limit
+    tr_id = str(uuid4())
+
+    class StubStorage:
+        def __init__(self):
+            self.storage_path = ConcurrencyLeaseStorage().storage_path
+            self.lease_id = uuid4()
+
+        async def list_holders_for_limit(self, limit_id):
+            if limit_id == v2.id:
+                # Return nested holder structure to ensure API supports both shapes
+                return [{"holder": {"type": "task_run", "id": tr_id}, "slots": 1}]
+            return []
+
+        async def read_active_lease_ids(self, limit: int = 100):
+            return [self.lease_id]
+
+        async def read_lease(self, lease_id):
+            if lease_id == self.lease_id:
+                return ResourceLease(
+                    resource_ids=[v2.id],
+                    expiration=datetime.now(timezone.utc) + timedelta(minutes=5),
+                    metadata=ConcurrencyLimitLeaseMetadata(
+                        slots=1, holder={"type": "task_run", "id": tr_id}
+                    ),
+                )
+            return None
+
+    # Patch the server module to use our stub
+    import prefect.server.api.concurrency_limits as v1api
+
+    monkeypatch.setattr(v1api, "get_concurrency_lease_storage", lambda: StubStorage())
```

Comment:
This stub storage shouldn't be necessary if you're using the filesystem lease storage, right?